src/Build.UnitTests/Graph/ProjectGraph_Tests.cs

```diff
@@ -37,6 +46,220 @@ public void TestGraphWithSingleNode()
                 projectNode.FullPath.ShouldBe(testProject.ProjectFile);
             }
         }
+
+        /// <summary>
+        ///  A  
+        /// / \ 
+        ///B   C
+        /// </summary>
+        [Fact]
+        public void TestGraphWithThreeNodes()
+        {
+            string projectAContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+                  <PropertyGroup>
+                    <TestProperty>value</TestProperty>
```

Comment:
These `TestProperty` properties don't seem necessary in any of these projects. It's OK for some of these projects to literally just be `<Project></Project>`